ecl/hqlcpp/hqlcset.cpp

```diff
@@ -1424,24 +1415,24 @@ void GeneralSetCursor::buildAssignSelect(BuildCtx & ctx, const CHqlBoundTarget &
     {
         OwnedHqlExpr select = createSelectExpr(LINK(dsIndexExpr), LINK(element));
         if (!cursor->isConditional())
-            translator.buildExprAssign(ctx, target, select);
+            translator.buildExprOrAssign(ctx, target, select, tgt);
         else
         {
             // if (row) tgt = x else tgt = dft;
             BuildCtx subctx(ctx);
```

Comment:
Unfortunately this code is not correct.  If it is building an assignment it will be correct.  However, if it is building an expression it will generate invalid code.  The reason is that buildExpr expects the result returned in tgt to be valid in the calling context.  With this code it will always be set to the value of null.

I suspect it isn't used very much - I would change the code to say

  if (target)
  {
    ..code..
  }
  else
    buildTempExpr(ctx, expr, *tgt);

Check regression suite before and after to see what files are effected and compare the code.